typed-racket-lib/typed-racket/infer/infer-unit.rkt

```diff
@@ -480,12 +480,14 @@
               (ValuesDots: (list (Result: _ psets _) ...) _ _))
           (AnyValues: q))
          (cset-join
```

Comment:
Is this still correct when there `psets` is non-empty?

Either way, I think this code would be easier to understand as an if-statement:

```
(if (null? psets)
  empty
  (cset-join ....))
```

Reply:
My understanding is that the empty cset (corresponding to ∅ in the theory) serves as an identity element to `cset-meet` so joining it should have no effect - but using an `if` instead will certainly save some unnecessary steps.